pkg/cmd/gist/create/create.go

```diff
@@ -176,6 +176,15 @@ func processFiles(stdin io.ReadCloser, filenameOverride string, filenames []stri
 			}
 			stdin.Close()
 		} else {
+			isBinary, err := shared.DetectedBinaryFileContents(f, false)
```

Comment:
Does detecting binary content advance the reader in `f`, which is going to again get used below in `ioutil.ReadFile(f)`? Should we reopen the file after this or otherwise rewind the reader?

Reply:
@mislav Yes, that's a redundant call. I shoudn't have done that. Now since our helper `DetectedBinaryFileContents` detects from file contents as well, and reading a file while creating a gist is inevitable, I now chose to read the file and then pass the contents itself to skip a redundant read call to the file. Now `gh gist create` reads the file only once. I've updated the PR.